extensions/amp-sticky-ad/0.1/amp-sticky-ad.js

```diff
@@ -37,6 +38,72 @@ class AmpStickyAd extends AMP.BaseElement {
       dev.warn(TAG, `TAG ${TAG} disabled`);
       return;
     }
+
+    const children = this.getRealChildren();
+    user.assert((children.length == 1 && children[0].tagName == 'AMP-AD'),
+        'amp-sticky-ad must have a single amp-ad child');
+    /** @const @private {!Element} */
+    this.ad_ = children[0];
+
+    /** @private @const {!Viewport} */
+    this.viewport_ = this.getViewport();
+
+    /** @const @private {!Vsync} */
+    this.vsync_ = this.getVsync();
+
+    /** @const @private {!Unlisten}
+     * On viewport scroll, check requirements for amp-stick-ad to display.
+     */
+    this.onScrollListener_ =
+        this.viewport_.onScroll(() => this.displayAfterScroll());
+  }
+
+  /** @override */
+  layoutCallback() {
+    this.isExperimentOn_ = isExperimentOn(this.getWin(), TAG);
+    if (!this.isExperimentOn_) {
+      dev.warn(TAG, `TAG ${TAG} disabled`);
+      return Promise.resolve();
+    }
+    return Promise.resolve();
+  }
+
+  /** @private
+   * The listener function that listen on onScroll event and
+   * show sticky ad when user scroll at least one viewport and
+   * there is at least one more viewport available.
+   */
+  displayAfterScroll() {
+    this.scrollTop_ = this.viewport_.getScrollTop();
+    this.scrollHeight_ = this.viewport_.getScrollHeight();
```

Comment:
It looks like we can move `scrollHeight` measurement into the `if` below. Then it will only be measured when the user has scrolled far enough, which could reduce number of expensive measures.